ruby/red-arrow/lib/arrow/slicer.rb

```diff
@@ -163,8 +163,41 @@ def reject(&block)
         RejectCondition.new(@column, block)
       end
 
-      def match_substring(substring, ignore_case: false)
-        MatchSubstringFamilyCondition.new("match_substring",
+      def end_with?(substring, ignore_case: false)
+        MatchSubstringFamilyCondition.new("ends_with",
+                                          @column, substring, ignore_case)
+      end
+
+      def match_like?(pattern, ignore_case: false)
+        MatchSubstringFamilyCondition.new("match_like",
+                                          @column, pattern, ignore_case)
+      end
+
+      def match_substring?(pattern, ignore_case: false)
+        case pattern
+        when String
+          MatchSubstringFamilyCondition.new("match_substring",
+                                            @column, pattern, ignore_case)
+        when Regexp
+          MatchSubstringFamilyCondition.new("match_substring_regex",
+                                            @column,
+                                            pattern.source,
+                                            ignore_case)
```

Comment:
Can we use `pattern.casefold?` as the default value of `ignore_case`?

Reply:
Thanks! It was a good lesson.